test/compilationTests/stringutils/strings.sol

```diff
@@ -506,15 +519,34 @@ library strings {
                     let mask := not(sub(exp(2, mul(8, sub(32, needlelen))), 1))
                     let needledata := and(mload(needleptr), mask)
                     ptr := add(selfptr, sub(selflen, needlelen))
-                    loop:
-                    jumpi(ret, eq(and(mload(ptr), mask), needledata))
-                    ptr := sub(ptr, 1)
-                    jumpi(loop, gt(add(ptr, 1), selfptr))
-                    ptr := selfptr
-                    jump(exit)
-                    ret:
-                    ptr := add(ptr, needlelen)
-                    exit:
+
+                    if iszero(eq(and(mload(ptr), mask), needledata)) {
```

Comment:
For the record: I think we need to have a close look at this one - I'm not sure whether the current code does what it should.

Reply:
yeah, I am aware of that and will look into this too. I'd highly appreciate others looking into it, too. :-)